languages/tree-sitter-stack-graphs-java/src/stack-graphs.tsg

```diff
@@ -0,0 +1,369 @@
+;;;;;;;;;;;;;;;;;;;
+;; Global Variables
+
+global FILE_PATH
+global ROOT_NODE
+global JUMP_TO_SCOPE_NODE
+
+;;;;;;;;;;;;;;;;;;;;;;;
+;; Attribute Shorthands
+
+attribute node_definition = node        => type = "pop_symbol", node_symbol = node, is_definition
+attribute node_reference = node         => type = "push_symbol", node_symbol = node, is_reference
+attribute pop_node = node               => type = "pop_symbol", node_symbol = node
+attribute pop_scoped_node = node        => type = "pop_scoped_symbol", node_symbol = node
+attribute pop_scoped_symbol = symbol    => type = "pop_scoped_symbol", symbol = symbol
+attribute pop_symbol = symbol           => type = "pop_symbol", symbol = symbol
+attribute push_node = node              => type = "push_symbol", node_symbol = node
+attribute push_scoped_node = node       => type = "push_scoped_symbol", node_symbol = node
+attribute push_scoped_symbol = symbol   => type = "push_scoped_symbol", symbol = symbol
+attribute push_symbol = symbol          => type = "push_symbol", symbol = symbol
+attribute scoped_node_definition = node => type = "pop_scoped_symbol", node_symbol = node, is_definition
+attribute scoped_node_reference = node  => type = "push_scoped_symbol", node_symbol = node, is_reference
+attribute symbol_definition = symbol    => type = "pop_symbol", symbol = symbol, is_definition
+attribute symbol_reference = symbol     => type = "push_symbol", symbol = symbol, is_reference
+
+attribute node_symbol = node            => symbol = (source-text node), source_node = node
+
+; ######
+; #     # #####   ####   ####  #####    ##   #    #  ####
+; #     # #    # #    # #    # #    #  #  #  ##  ## #
+; ######  #    # #    # #      #    # #    # # ## #  ####
+; #       #####  #    # #  ### #####  ###### #    #      #
+; #       #   #  #    # #    # #   #  #    # #    # #    #
+; #       #    #  ####   ####  #    # #    # #    #  ####
+;
+; ########################################################
+
+(program)@prog {
+  node @prog.defs
+  node @prog.lexical_scope
+  edge @prog.lexical_scope -> ROOT_NODE
+  edge ROOT_NODE -> @prog.defs
+}
+
+(program (_)@declaration)@prog {
+  edge @prog.defs -> @declaration.defs
+  edge @declaration.lexical_scope -> @prog.lexical_scope
+}
+
+
+;; =======================
+;;  Top level declarations
+;; =======================
+
+(import_declaration) @import {
+  node @import.defs
+  node @import.lexical_scope
+}
+
+(import_declaration
+  (scoped_identifier
+    name: (identifier) @scope_name)
+  ) @import {
+
+  node @scope_name.value
+  attr (@scope_name.value) node_definition = @scope_name
+
+  edge @import.defs -> @scope_name.value
+}
+
+(import_declaration
+  (scoped_identifier
+    scope: (scoped_identifier
```

Comment:
There can be N amount of nested `scoped_identifier` here... note to myself/TODO: expand this so that it can apply to any nested scopes.

ie.
`import test.Item` vs `import java.util.Collection` vs `import test.foo.bar.baz.Item`

Reply:
I added a test for this and realized that it actually does work for N nested scopes. 🎉 I thought I would need a `+` pattern matcher for nested `scoped_identifiers` but turns out it works without it.